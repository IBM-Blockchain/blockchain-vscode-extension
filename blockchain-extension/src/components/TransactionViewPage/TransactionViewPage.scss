@import '@carbon/layout/scss/_spacing';
@import '@carbon/grid/scss/grid';

body {
    font-size: 14px;
    letter-spacing: 0.01px;
    font-weight: 400;
    -webkit-font-smoothing: antialiased;
    -moz-font-smoothing: antialiased;
    -o-font-smoothing: antialiased;
}

.page-container {
    margin: 0;
    padding: 0;
    h5 {
        padding-bottom: $spacing-06;
    }
}

.inner-container {
    height: 100vh;
}

.page-contents {
    padding: $spacing-07;
}

.titles-container {
    padding: 0px $spacing-04;

    h1 {
        padding: $spacing-09 0px;
    }
}

.contents-container {
    margin: $spacing-02;
    padding: 0px $spacing-04;
}

.contents-left {

    h5 {
        padding-top: $spacing-09;
    }
}
