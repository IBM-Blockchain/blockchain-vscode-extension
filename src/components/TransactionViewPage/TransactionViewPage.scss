@import '@carbon/layout/scss/_spacing';
@import '@carbon/grid/scss/grid';
@import '@carbon/colors/scss/colors';

body {
    font-size: 14px;
    letter-spacing: 0.01px;
    font-weight: 400;
    -webkit-font-smoothing: antialiased;
    -moz-font-smoothing: antialiased;
    -o-font-smoothing: antialiased;
}

.page-container {
    margin: 0;
    padding: 0;

    h5 {
        padding-bottom: $spacing-06;
    }
}

.inner-container {
    height: 100vh;
    max-width: 100%;
}

.page-contents {
    padding: $spacing-07;
}

.titles-container {
    padding: 0px $spacing-04;

    h1 {
        padding: $spacing-09 0px;
    }
}

.contents-container {
    margin: $spacing-02;
}

.contents-left {

    h5 {
        padding-top: $spacing-09;
    }
}

.smart-contract-item {
    padding-top: $spacing-02;

    &:first-of-type {
        padding-top: $spacing-01;
    }
}

.clickable-smart-contract {
    color: $carbon--blue-50;

    &:hover {
        cursor: pointer;
    }
}

.disabled-smart-contract {
    color: $carbon--gray-50;
}
