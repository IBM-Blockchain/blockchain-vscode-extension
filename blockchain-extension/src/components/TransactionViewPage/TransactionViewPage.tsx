import React, { Component } from 'react';
import './TransactionViewPage.scss';
import Sidebar from '../TransactionViewSidebar/TransactionViewSidebar';
import { Accordion, AccordionItem, Link } from 'carbon-components-react';

type PageProps = {
    activeSmartContract: string
};

type PageState = { 
    activeSmartContract: string
}

class TransactionViewPage extends Component<PageProps, PageState> {

    constructor(props: Readonly<PageProps>) {
        super(props);
        this.state = {
            activeSmartContract: this.props.activeSmartContract
        }
    }

    public render() {
        return (
            <div className="page-container bx--grid">
                <div className="inner-container bx--row">
                    <Sidebar/>
                    <div className="page-contents bx--col">
                        <div className="titles-container">
                            <Link>{this.state.activeSmartContract}</Link>
                            <h1>Transactions</h1>
                        </div>
                        <div className="contents-container bx--row">
                            <div className="contents-left bx--col">
                                <p>Welcome to your transaction web view for <Link>{this.props.activeSmartContract}</Link>; use this to submit and evaluate new and existing transactions.</p>
                                <p>Unsure of where to start? Follow our introductory Getting Started section opposite.</p>
                                <br/>
                                <h5>Smart Contracts</h5>
                                <p>To switch to the web view for a different smart contract, select the smart contract below</p>
                                <Link>{this.state.activeSmartContract}</Link>
                                <br/>
                                <Link>congaContract@0.0.1</Link>
                            </div>
                            <div className="contents-right bx--col" id="contents-right">
                                <h5>Getting Started</h5>
                                <Accordion>
                                    <AccordionItem title="Create a new transaction"></AccordionItem>
                                </Accordion>
                                <Accordion>
                                    <AccordionItem title="Edit existing transactions"></AccordionItem>
                                </Accordion>
                                <Accordion>
                                    <AccordionItem title="Favouriting & deleting transactions"></AccordionItem>
                                </Accordion>
                                <Accordion>
                                    <AccordionItem title="Submitting and evaluating: what's the difference?"></AccordionItem>
                                </Accordion>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default TransactionViewPage;
