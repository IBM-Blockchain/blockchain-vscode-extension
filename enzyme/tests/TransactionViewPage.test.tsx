import React from 'react';
import renderer from 'react-test-renderer';
import TransactionViewPage from '../../src/components/TransactionViewPage/TransactionViewPage';
import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
chai.should();
chai.use(sinonChai);

describe('TransactionViewPage component', () => {

    let mySandBox: sinon.SinonSandbox;

    beforeEach(async () => {
        mySandBox = sinon.createSandbox();
    });

    afterEach(async () => {
        mySandBox.restore();
    });

    it('should render the expected snapshot', async () => {
        const component: any = renderer
            .create(<TransactionViewPage activeSmartContract="penguinContract@0.0.1"/>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });   

});
