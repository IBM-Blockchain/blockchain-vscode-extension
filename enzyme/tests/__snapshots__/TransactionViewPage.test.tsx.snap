// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionViewPage component should render the expected snapshot 1`] = `
<div
  className="page-container bx--grid"
  data-test-id="txn-page"
>
  <div
    className="inner-container bx--row"
  >
    <div
      className="sidebar-container"
      id="sidebar-container"
    >
      <div
        className="panel-container"
      >
        <button
          className="bx--btn bx--btn--field bx--btn--primary"
          disabled={false}
          id="create-button"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          Create a new transaction
        </button>
        <button
          className="bx--btn bx--btn--field bx--btn--primary"
          disabled={false}
          id="import-button"
          onClick={[Function]}
          tabIndex={0}
          type="button"
        >
          Import existing transaction
        </button>
      </div>
      <div
        className="panel-container"
      >
        <h5>
          Filters
        </h5>
      </div>
      <div
        className="panel-container"
      >
        <h5>
          Transactions
        </h5>
      </div>
    </div>
    <div
      className="page-contents bx--col"
    >
      <div
        className="titles-container"
      >
        <span>
          greenContract@0.0.1
        </span>
        <h1>
          Transactions
        </h1>
      </div>
      <div
        className="contents-container bx--row"
      >
        <div
          className="contents-left bx--col"
        >
          <p>
            Welcome to your transaction web view for 
            <span
              className="disabled-smart-contract"
            >
              greenContract@0.0.1
            </span>
            ; use this to submit and evaluate new and existing transactions.
          </p>
          <p>
            Unsure of where to start? Follow our introductory Getting Started section opposite.
          </p>
          <br />
          <h5>
            Smart Contracts
          </h5>
          <p>
            To switch to the web view for a different smart contract, select the smart contract below
          </p>
          <ul>
            <li
              className="smart-contract-item disabled-smart-contract"
            >
              greenContract@0.0.1
            </li>
            <li
              className="smart-contract-item clickable-smart-contract"
              onClick={[Function]}
            >
              blueContract@0.0.1
            </li>
          </ul>
        </div>
        <div
          className="contents-right bx--col"
          id="contents-right"
        >
          <h5>
            Getting Started
          </h5>
          <ul
            className="bx--accordion"
          >
            <li
              className="bx--accordion__item"
              onAnimationEnd={[Function]}
            >
              <button
                aria-expanded={false}
                className="bx--accordion__heading"
                onClick={[Function]}
                onKeyDown={[Function]}
                title="Expand/Collapse"
                type="button"
              >
                <svg
                  aria-label="Expand/Collapse"
                  className="bx--accordion__arrow"
                  focusable="false"
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 8l-5 5-.7-.7L9.6 8 5.3 3.7 6 3z"
                  />
                </svg>
                <div
                  className="bx--accordion__title"
                >
                  Create a new transaction
                </div>
              </button>
              <div
                className="bx--accordion__content"
              />
            </li>
          </ul>
          <ul
            className="bx--accordion"
          >
            <li
              className="bx--accordion__item"
              onAnimationEnd={[Function]}
            >
              <button
                aria-expanded={false}
                className="bx--accordion__heading"
                onClick={[Function]}
                onKeyDown={[Function]}
                title="Expand/Collapse"
                type="button"
              >
                <svg
                  aria-label="Expand/Collapse"
                  className="bx--accordion__arrow"
                  focusable="false"
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 8l-5 5-.7-.7L9.6 8 5.3 3.7 6 3z"
                  />
                </svg>
                <div
                  className="bx--accordion__title"
                >
                  Edit existing transactions
                </div>
              </button>
              <div
                className="bx--accordion__content"
              />
            </li>
          </ul>
          <ul
            className="bx--accordion"
          >
            <li
              className="bx--accordion__item"
              onAnimationEnd={[Function]}
            >
              <button
                aria-expanded={false}
                className="bx--accordion__heading"
                onClick={[Function]}
                onKeyDown={[Function]}
                title="Expand/Collapse"
                type="button"
              >
                <svg
                  aria-label="Expand/Collapse"
                  className="bx--accordion__arrow"
                  focusable="false"
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 8l-5 5-.7-.7L9.6 8 5.3 3.7 6 3z"
                  />
                </svg>
                <div
                  className="bx--accordion__title"
                >
                  Favouriting & deleting transactions
                </div>
              </button>
              <div
                className="bx--accordion__content"
              />
            </li>
          </ul>
          <ul
            className="bx--accordion"
          >
            <li
              className="bx--accordion__item"
              onAnimationEnd={[Function]}
            >
              <button
                aria-expanded={false}
                className="bx--accordion__heading"
                onClick={[Function]}
                onKeyDown={[Function]}
                title="Expand/Collapse"
                type="button"
              >
                <svg
                  aria-label="Expand/Collapse"
                  className="bx--accordion__arrow"
                  focusable="false"
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  style={
                    Object {
                      "willChange": "transform",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 8l-5 5-.7-.7L9.6 8 5.3 3.7 6 3z"
                  />
                </svg>
                <div
                  className="bx--accordion__title"
                >
                  Submitting and evaluating: what's the difference?
                </div>
              </button>
              <div
                className="bx--accordion__content"
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
